spec/Section 7 -- Response.md

```diff
@@ -135,14 +145,103 @@ locations, where each location is a map with the keys `line` and `column`, both
 positive numbers starting from `1` which describe the beginning of an
 associated syntax element.
 
+If an error can be associated to a particular field in the GraphQL result, it
+must contain an entry with the key `path` that details the path of the
+response field which experienced the error. This allows clients to identify
+whether a `null` result is intentional or caused by a runtime error.
+
+This field should be a list of path segments starting at the root of the
+response and ending with the field associated with the error. Path segments
+that represent fields should be strings, and path segments that
+represent list indices should be 0-indexed integers. If the error happens
+in an aliased field, the path to the error should use the aliased name, since
+it represents a path in the response, not in the query.
+
+For example, if fetching one of the friends' names fails in the following
+query:
+
+```graphql
+{
+  hero(episode: $episode) {
+    name
+    heroFriends: friends {
+      id
+      name
+    }
+  }
+}
+```
+
+The response might look like:
+
+```js
+{
+  "data": {
+    "hero": {
+      "name": "R2-D2",
+      "heroFriends": [
+        {
+          "id": "1000",
+          "name": "Luke Skywalker"
+        },
+        {
+          "id": "1002",
+          "name": null
+        },
+        {
+          "id": "1003",
+          "name": "Leia Organa"
+        }
+      ]
+    }
+  },
+  "errors": [
+    {
+      "message": "Name for character with ID 1002 could not be fetched.",
+      "locations": [ { "line": 6, "column": 7 } ],
+      "path": [ "hero", "heroFriends", 1, "name" ]
+    }
+  ]
+}
+```
+
+If the field which experienced an error was declared as `Non-Null`, the `null`
+result will bubble up to the next nullable field. In that case, the `path`
+for the error should include the full path to the result field where the error
+occurred, even if that field is not present in the response.
```

Comment:
Yes that is the intention - the error path includes path segments that go deeper than the actual JSON the client returned. If the client doesn't need those path segments it can ignore them, and the server doesn't really need to do any extra work to make it happen in standard execution.